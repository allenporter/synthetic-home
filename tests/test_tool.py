"""Tests for the sythethic-home `create_inventory` command."""

import shlex
import subprocess
import asyncio
import logging
import pathlib

import pytest

from syrupy.assertion import SnapshotAssertion


_LOGGER = logging.getLogger(__name__)

BIN = "synthetic-home"

TEST_HOMES = pathlib.Path("tests/homes")


async def run(cmd: list[str], stdin: bytes | None = None) -> bytes:
    """Run the command, returning stdout."""
    cmd_string = " ".join([shlex.quote(arg) for arg in cmd])
    _LOGGER.debug("Running command: %s", cmd_string)
    proc = await asyncio.create_subprocess_shell(
        cmd_string,
        stdin=subprocess.PIPE,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
    )
    out, err = await proc.communicate(stdin)
    if proc.returncode:
        errors = [f"Command '{cmd_string}' failed with return code {proc.returncode}"]
        if out:
            errors.append(out.decode("utf-8"))
        if err:
            errors.append(err.decode("utf-8"))
        _LOGGER.debug("\n".join(errors))
        raise ValueError("\n".join(errors))
    return out


@pytest.mark.parametrize(
    ("home_filename"),
    list(TEST_HOMES.glob("*.yaml")),
    ids=[str(filename) for filename in TEST_HOMES.glob("*.yaml")],
)
async def test_build(home_filename: pathlib.Path, snapshot: SnapshotAssertion) -> None:
    """Test build commands."""
    result = await run([BIN, "create_inventory", str(home_filename)])
    str_result = result.decode("utf-8")
    assert str_result == snapshot


async def test_list_device_types(snapshot: SnapshotAssertion) -> None:
    """Test build commands."""
    result = await run([BIN, "list_device_types"])
    str_result = result.decode("utf-8")
    assert str_result == snapshot
