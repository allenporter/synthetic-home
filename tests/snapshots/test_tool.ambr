# serializer version: 1
# name: test_build[tests/homes/alarm-system-example.yaml]
  '''
  ---
  areas:
  - name: Entryway
    id: entryway
  devices:
  - name: Alarm
    id: alarm
    area: entryway
    info:
      manufacturer: Ring
  entities:
  - name: Alarm
    id: alarm_control_panel.alarm
    area: entryway
    device: alarm
    state: disarmed
    attributes:
      code_format: number
      supported_features:
      - alarm_control_panel.AlarmControlPanelEntityFeature.ARM_HOME
      - alarm_control_panel.AlarmControlPanelEntityFeature.ARM_AWAY
      - alarm_control_panel.AlarmControlPanelEntityFeature.TRIGGER
  - name: Alarm Tamper
    id: binary_sensor.alarm_tamper
    area: entryway
    device: alarm
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.TAMPER
  - name: Alarm
    id: text.alarm
    area: entryway
    device: alarm
    attributes:
      mode: text
  - name: Alarm
    id: siren.alarm
    area: entryway
    device: alarm
    state: 'off'
    attributes:
      supported_features:
      - siren.SirenEntityFeature.VOLUME_SET
  
  
  '''
# ---
# name: test_build[tests/homes/calendar-example.yaml]
  '''
  ---
  areas: []
  devices:
  - name: Calendar
    id: calendar
  entities:
  - name: Calendar
    id: calendar.calendar
    device: calendar
    attributes:
      supported_features:
      - calendar.CalendarEntityFeature.CREATE_EVENT
      - calendar.CalendarEntityFeature.DELETE_EVENT
      - calendar.CalendarEntityFeature.UPDATE_EVENT
  
  
  '''
# ---
# name: test_build[tests/homes/camera-example.yaml]
  '''
  ---
  areas:
  - name: Backyard
    id: backyard
  devices:
  - name: Outdoor Camera
    id: outdoor_camera
    area: backyard
    info:
      model: Spotlight Cam Battery
      manufacturer: Ring
      sw_version: 2.4.1
  entities:
  - name: Outdoor Camera
    id: camera.outdoor_camera
    area: backyard
    device: outdoor_camera
    attributes:
      supported_features:
      - camera.CameraEntityFeature.ON_OFF
      - camera.CameraEntityFeature.STREAM
  - name: Outdoor Camera Motion
    id: binary_sensor.outdoor_camera_motion
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.MOTION
  - name: Outdoor Camera Person
    id: binary_sensor.outdoor_camera_person
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.OCCUPANCY
  - name: Outdoor Camera Sound
    id: binary_sensor.outdoor_camera_sound
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.SOUND
  
  
  '''
# ---
# name: test_build[tests/homes/display-led-example.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Led Display
    id: led_display
    area: family_room
  entities:
  - name: Led Display
    id: light.led_display
    area: family_room
    device: led_display
    attributes:
      supported_color_modes:
      - brightness
      color_mode: brightness
      brightness: 100
  - name: Led Display
    id: select.led_display
    area: family_room
    device: led_display
    state: blink
    attributes:
      options:
      - scroll
      - static
      - blink
  - name: Led Display
    id: text.led_display
    area: family_room
    device: led_display
    attributes:
      mode: text
  
  
  '''
# ---
# name: test_build[tests/homes/door-sensor-example.yaml]
  '''
  ---
  areas:
  - name: Front door
    id: front_door
  devices:
  - name: Front Door Sensor
    id: front_door_sensor
    area: front_door
    info:
      manufacturer: Aqara
  entities:
  - name: Front Door Sensor
    id: binary_sensor.front_door_sensor
    area: front_door
    device: front_door_sensor
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.DOOR
  - name: Front Door Sensor Battery
    id: binary_sensor.front_door_sensor_battery
    area: front_door
    device: front_door_sensor
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.BATTERY
  - name: Front Door Sensor Battery
    id: sensor.front_door_sensor_battery
    area: front_door
    device: front_door_sensor
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/fan-example.yaml]
  '''
  ---
  areas:
  - name: Kitchen
    id: kitchen
  devices:
  - name: Counter Fan
    id: counter_fan
    area: kitchen
  entities:
  - name: Counter Fan
    id: fan.counter_fan
    area: kitchen
    device: counter_fan
    state: 'off'
    attributes:
      supported_features:
      - fan.FanEntityFeature.OSCILLATE
      - fan.FanEntityFeature.SET_SPEED
      oscillating: false
  
  
  '''
# ---
# name: test_build[tests/homes/garage-door-example.yaml]
  '''
  ---
  areas:
  - name: Front Yard
    id: front_yard
  devices:
  - name: Garage Door
    id: garage_door
    area: front_yard
  entities:
  - name: Garage Door
    id: cover.garage_door
    area: front_yard
    device: garage_door
    state: false
    attributes:
      device_class: cover.CoverDeviceClass.GARAGE
      supported_features:
      - cover.CoverEntityFeature.OPEN
      - cover.CoverEntityFeature.CLOSE
  - name: Garage Door
    id: light.garage_door
    area: front_yard
    device: garage_door
    state: false
    attributes:
      supported_color_modes:
      - onoff
      color_mode:
      - onoff
  
  
  '''
# ---
# name: test_build[tests/homes/gate-example.yaml]
  '''
  ---
  areas:
  - name: Front Yard
    id: front_yard
  devices:
  - name: Driveway Gate
    id: driveway_gate
    area: front_yard
  entities:
  - name: Driveway Gate
    id: cover.driveway_gate
    area: front_yard
    device: driveway_gate
    state: closed
    attributes:
      device_class: cover.CoverDeviceClass.GATE
      supported_features:
      - cover.CoverEntityFeature.OPEN
      - cover.CoverEntityFeature.CLOSE
  
  
  '''
# ---
# name: test_build[tests/homes/heat-pump-example.yaml]
  '''
  ---
  areas:
  - name: Family room
    id: family_room
  devices:
  - name: Family Room
    id: family_room
    area: family_room
    info:
      model: Thermostat
      manufacturer: Nest
      sw_version: 1.0.0
  entities:
  - name: Family Room
    id: climate.family_room
    area: family_room
    device: family_room
    attributes:
      unit_of_measurement: "\xB0F"
      hvac_modes:
      - heat
      - 'off'
      supported_features:
      - climate.ClimateEntityFeature.FAN_MODE
      - climate.ClimateEntityFeature.TURN_ON
      - climate.ClimateEntityFeature.TURN_OFF
      hvac_mode: 'off'
      hvac_action: 'off'
      current_temperature: 60
      target_temperature: 60
  - name: Family Room Temperature
    id: sensor.family_room_temperature
    area: family_room
    device: family_room
    state: '60'
    attributes:
      native_unit_of_measurement: "\xB0C"
      device_class: sensor.SensorDeviceClass.TEMPERATURE
      state_class: sensor.SensorStateClass.MEASUREMENT
  
  
  '''
# ---
# name: test_build[tests/homes/home1.yaml]
  '''
  ---
  areas:
  - name: Living Room
    id: living_room
  - name: Backyard
    id: backyard
  - name: Bedroom 1
    id: bedroom_1
  - name: Kitchen
    id: kitchen
  - name: Bathroom
    id: bathroom
  devices:
  - name: Fireplace Light
    id: fireplace_light
    area: living_room
    info:
      model: Smart LED Bulb
      manufacturer: Philips
      sw_version: 1.2.3
  - name: Thermostat
    id: thermostat
    area: living_room
    info:
      model: Ecobee4 Smart Thermostat
      manufacturer: Ecobee
      sw_version: 4.0.0
  - name: Outdoor Camera
    id: outdoor_camera
    area: backyard
    info:
      model: Spotlight Cam Battery
      manufacturer: Ring
      sw_version: 2.4.1
  - name: Bedroom 1 Light
    id: bedroom_1_light
    area: bedroom_1
    info:
      model: Smart LED Bulb
      manufacturer: Philips
      sw_version: 1.2.3
  - name: Kitchen Light
    id: kitchen_light
    area: kitchen
    info:
      manufacturer: Philips
      sw_version: 2.47.8
  - name: Coffee Maker
    id: coffee_maker
    area: kitchen
    info:
      model: S31 Lite zb
      manufacturer: SONOFF
      sw_version: 2.3.3.140
  - name: Bathroom Light
    id: bathroom_light
    area: bathroom
    info:
      model: Smart LED Bulb
      manufacturer: Philips
      sw_version: 1.2.3
  entities:
  - name: Fireplace Light
    id: light.fireplace_light
    area: living_room
    device: fireplace_light
    attributes:
      supported_color_modes:
      - brightness
      color_mode: brightness
      brightness: 100
  - name: Thermostat
    id: climate.thermostat
    area: living_room
    device: thermostat
    attributes:
      unit_of_measurement: "\xB0C"
      supported_features:
      - climate.ClimateEntityFeature.FAN_MODE
      - climate.ClimateEntityFeature.TURN_ON
      - climate.ClimateEntityFeature.TURN_OFF
      - climate.ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
      hvac_modes:
      - 'off'
      - cool
      - heat
      - auto
      hvac_mode: 'off'
      hvac_action: 'off'
      current_temperature: 22
      target_temperature: 22
  - name: Thermostat Temperature
    id: sensor.thermostat_temperature
    area: living_room
    device: thermostat
    attributes:
      native_unit_of_measurement: "\xB0C"
      device_class: sensor.SensorDeviceClass.TEMPERATURE
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_value: 22
  - name: Thermostat Humidity
    id: sensor.thermostat_humidity
    area: living_room
    device: thermostat
    attributes:
      native_unit_of_measurement: '%'
      device_class: sensor.SensorDeviceClass.HUMIDITY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_value: 45
  - name: Outdoor Camera
    id: camera.outdoor_camera
    area: backyard
    device: outdoor_camera
    attributes:
      supported_features:
      - camera.CameraEntityFeature.ON_OFF
      - camera.CameraEntityFeature.STREAM
  - name: Outdoor Camera Motion
    id: binary_sensor.outdoor_camera_motion
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.MOTION
  - name: Outdoor Camera Person
    id: binary_sensor.outdoor_camera_person
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.OCCUPANCY
  - name: Outdoor Camera Sound
    id: binary_sensor.outdoor_camera_sound
    area: backyard
    device: outdoor_camera
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.SOUND
  - name: Bedroom 1 Light
    id: light.bedroom_1_light
    area: bedroom_1
    device: bedroom_1_light
    state: 'off'
    attributes:
      supported_color_modes:
      - onoff
      color_mode: onoff
  - name: Kitchen Light
    id: light.kitchen_light
    area: kitchen
    device: kitchen_light
    state: 'off'
    attributes:
      supported_color_modes:
      - onoff
      color_mode: onoff
  - name: Coffee Maker Energy
    id: sensor.coffee_maker_energy
    area: kitchen
    device: coffee_maker
    state: '1'
    attributes:
      device_class: sensor.SensorDeviceClass.ENERGY
      state_class: sensor.SensorStateClass.TOTAL_INCREASING
      native_unit_of_measurement: kWh
  - name: Coffee Maker
    id: switch.coffee_maker
    area: kitchen
    device: coffee_maker
    state: true
    attributes:
      device_class: switch.SwitchDeviceClass.OUTLET
  - name: Bathroom Light
    id: light.bathroom_light
    area: bathroom
    device: bathroom_light
    state: 'off'
    attributes:
      supported_color_modes:
      - onoff
      color_mode: onoff
  
  
  '''
# ---
# name: test_build[tests/homes/hvac-example.yaml]
  '''
  ---
  areas:
  - name: Family room
    id: family_room
  devices:
  - name: Family Room
    id: family_room
    area: family_room
    info:
      model: Thermostat
      manufacturer: Nest
      sw_version: 1.0.0
  entities:
  - name: Family Room
    id: climate.family_room
    area: family_room
    device: family_room
    attributes:
      unit_of_measurement: "\xB0F"
      supported_features:
      - climate.ClimateEntityFeature.FAN_MODE
      - climate.ClimateEntityFeature.TURN_ON
      - climate.ClimateEntityFeature.TURN_OFF
      - climate.ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
      hvac_modes:
      - 'off'
      - cool
      - heat
      - auto
      hvac_mode: 'off'
      hvac_action: 'off'
      current_temperature: 62
      target_temperature: 22
  - name: Family Room Temperature
    id: sensor.family_room_temperature
    area: family_room
    device: family_room
    attributes:
      native_unit_of_measurement: "\xB0F"
      device_class: sensor.SensorDeviceClass.TEMPERATURE
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_value: 62
  - name: Family Room Humidity
    id: sensor.family_room_humidity
    area: family_room
    device: family_room
    attributes:
      native_unit_of_measurement: '%'
      device_class: sensor.SensorDeviceClass.HUMIDITY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_value: 45
  
  
  '''
# ---
# name: test_build[tests/homes/light-dimmable.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Family Room
    id: family_room
    area: family_room
  entities:
  - name: Family Room
    id: light.family_room
    area: family_room
    device: family_room
    attributes:
      supported_color_modes:
      - brightness
      color_mode: brightness
      brightness: 30
  
  
  '''
# ---
# name: test_build[tests/homes/light-example.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Family Room
    id: family_room
    area: family_room
  entities:
  - name: Family Room
    id: light.family_room
    area: family_room
    device: family_room
    state: 'off'
    attributes:
      supported_color_modes:
      - onoff
      color_mode: onoff
  
  
  '''
# ---
# name: test_build[tests/homes/light-rgbw.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Family Room
    id: family_room
    area: family_room
  entities:
  - name: Family Room
    id: light.family_room
    area: family_room
    device: family_room
    attributes:
      supported_color_modes:
      - rgbw
      color_mode: rgbw
      rgbw_color:
      - 255
      - 255
      - 0
      - 255
  
  
  '''
# ---
# name: test_build[tests/homes/mobile-phone-example.yaml]
  '''
  ---
  areas: []
  devices:
  - name: Android
    id: android
  - name: Iphone
    id: iphone
  entities:
  - name: Android
    id: notify.android
    device: android
  - name: Android
    id: device_tracker.android
    device: android
    state: home
  - name: Android Battery
    id: sensor.android_battery
    device: android
    state: '70'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  - name: Iphone
    id: notify.iphone
    device: iphone
  - name: Iphone
    id: device_tracker.iphone
    device: iphone
    state: home
  - name: Iphone Battery
    id: sensor.iphone_battery
    device: iphone
    state: '70'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/motion-sensor-example.yaml]
  '''
  ---
  areas:
  - name: Front Yard
    id: front_yard
  devices:
  - name: Front Yard Motion
    id: front_yard_motion
    area: front_yard
  entities:
  - name: Front Yard Motion
    id: binary_sensor.front_yard_motion
    area: front_yard
    device: front_yard_motion
    state: true
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.MOTION
  - name: Front Yard Motion Battery
    id: binary_sensor.front_yard_motion_battery
    area: front_yard
    device: front_yard_motion
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.BATTERY
  - name: Front Yard Motion Battery
    id: sensor.front_yard_motion_battery
    area: front_yard
    device: front_yard_motion
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/smart-blinds-example.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Left Shade
    id: left_shade
    area: family_room
  - name: Right Shade
    id: right_shade
    area: family_room
  entities:
  - name: Left Shade
    id: cover.left_shade
    area: family_room
    device: left_shade
    state: closed
    attributes:
      device_class: cover.CoverDeviceClass.BLIND
      supported_features:
      - cover.CoverEntityFeature.OPEN
      - cover.CoverEntityFeature.CLOSE
      - cover.CoverEntityFeature.SET_POSITION
  - name: Left Shade Battery
    id: sensor.left_shade_battery
    area: family_room
    device: left_shade
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  - name: Right Shade
    id: cover.right_shade
    area: family_room
    device: right_shade
    state: closed
    attributes:
      device_class: cover.CoverDeviceClass.BLIND
      supported_features:
      - cover.CoverEntityFeature.OPEN
      - cover.CoverEntityFeature.CLOSE
      - cover.CoverEntityFeature.SET_POSITION
  - name: Right Shade Battery
    id: sensor.right_shade_battery
    area: family_room
    device: right_shade
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/smart-lock-example.yaml]
  '''
  ---
  areas:
  - name: Front door
    id: front_door
  devices:
  - name: Front Door Lock
    id: front_door_lock
    area: front_door
    info:
      manufacturer: Yale
  entities:
  - name: Front Door Lock
    id: lock.front_door_lock
    area: front_door
    device: front_door_lock
    state: locked
  - name: Front Door Lock
    id: binary_sensor.front_door_lock
    area: front_door
    device: front_door_lock
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.LOCK
  - name: Front Door Lock Tamper
    id: binary_sensor.front_door_lock_tamper
    area: front_door
    device: front_door_lock
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.TAMPER
  - name: Front Door Lock Battery
    id: binary_sensor.front_door_lock_battery
    area: front_door
    device: front_door_lock
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.BATTERY
  - name: Front Door Lock Battery
    id: sensor.front_door_lock_battery
    area: front_door
    device: front_door_lock
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/smart-plug-example.yaml]
  '''
  ---
  areas:
  - name: Family room
    id: family_room
  devices:
  - name: Floor Lamp
    id: floor_lamp
    area: family_room
    info:
      manufacturer: Wyze
  entities:
  - name: Floor Lamp Energy
    id: sensor.floor_lamp_energy
    area: family_room
    device: floor_lamp
    state: '1'
    attributes:
      device_class: sensor.SensorDeviceClass.ENERGY
      state_class: sensor.SensorStateClass.TOTAL_INCREASING
      native_unit_of_measurement: kWh
  - name: Floor Lamp
    id: switch.floor_lamp
    area: family_room
    device: floor_lamp
    state: true
    attributes:
      device_class: switch.SwitchDeviceClass.OUTLET
  
  
  '''
# ---
# name: test_build[tests/homes/smart-speaker-example.yaml]
  '''
  ---
  areas:
  - name: Living Room
    id: living_room
  devices:
  - name: Smart Speaker
    id: smart_speaker
    area: living_room
    info:
      model: Echo Dot (4th Gen)
      manufacturer: Amazon
      sw_version: 4.5.1
  entities:
  - name: Smart Speaker
    id: media_player.smart_speaker
    area: living_room
    device: smart_speaker
    state: playing
    attributes:
      device_class: media_player.MediaPlayerDeviceClass.SPEAKER
      supported_features:
      - media_player.MediaPlayerEntityFeature.PLAY
      - media_player.MediaPlayerEntityFeature.PAUSE
      - media_player.MediaPlayerEntityFeature.STOP
      - media_player.MediaPlayerEntityFeature.VOLUME_STEP
      - media_player.MediaPlayerEntityFeature.VOLUME_MUTE
      - media_player.MediaPlayerEntityFeature.VOLUME_SET
      - media_player.MediaPlayerEntityFeature.TURN_ON
      - media_player.MediaPlayerEntityFeature.TURN_OFF
      - media_player.MediaPlayerEntityFeature.NEXT_TRACK
      - media_player.MediaPlayerEntityFeature.PREVIOUS_TRACK
      - media_player.MediaPlayerEntityFeature.PLAY_MEDIA
  
  
  '''
# ---
# name: test_build[tests/homes/switch-example.yaml]
  '''
  ---
  areas:
  - name: Chicken Coop
    id: chicken_coop
  devices:
  - name: Smart Feeder
    id: smart_feeder
    area: chicken_coop
  entities:
  - name: Smart Feeder
    id: switch.smart_feeder
    area: chicken_coop
    device: smart_feeder
    state: 'on'
    attributes:
      device_class: switch.SwitchDeviceClass.SWITCH
  
  
  '''
# ---
# name: test_build[tests/homes/todo-list-example.yaml]
  '''
  ---
  areas: []
  devices:
  - name: Tasks
    id: tasks
  entities:
  - name: Tasks
    id: todo.tasks
    device: tasks
    state: '2'
    attributes:
      supported_features:
      - todo.TodoListEntityFeature.CREATE_TODO_ITEM
      - todo.TodoListEntityFeature.DELETE_TODO_ITEM
      todo_items:
      - summary: Homework
      - summary: Call mom
  
  
  '''
# ---
# name: test_build[tests/homes/vacuum-example.yaml]
  '''
  ---
  areas:
  - name: Kitchen
    id: kitchen
  devices:
  - name: Robot Vacuum
    id: robot_vacuum
    area: kitchen
  entities:
  - name: Robot Vacuum
    id: vacuum.robot_vacuum
    area: kitchen
    device: robot_vacuum
    state: docked
    attributes:
      supported_features:
      - vacuum.VacuumEntityFeature.START
      - vacuum.VacuumEntityFeature.STOP
      - vacuum.VacuumEntityFeature.STATE
      - vacuum.VacuumEntityFeature.PAUSE
      - vacuum.VacuumEntityFeature.RETURN_HOME
  
  
  '''
# ---
# name: test_build[tests/homes/valve-example.yaml]
  '''
  ---
  areas:
  - name: Garden
    id: garden
  devices:
  - name: Back Yard Water Valve
    id: back_yard_water_valve
    area: garden
  entities:
  - name: Back Yard Water Valve
    id: valve.back_yard_water_valve
    area: garden
    device: back_yard_water_valve
    attributes:
      reports_position: true
      supported_features:
      - valve.ValveEntityFeature.OPEN
      - valve.ValveEntityFeature.CLOSE
      - valve.ValveEntityFeature.SET_POSITION
      current_valve_position: 0
  - name: Back Yard Water Valve Meter-reading
    id: sensor.back_yard_water_valve_meter_reading
    area: garden
    device: back_yard_water_valve
    state: '2000'
    attributes:
      native_unit_of_measurement: "m\xB3"
      device_class: sensor.SensorDeviceClass.WATER
      state_class: sensor.SensorStateClass.TOTAL_INCREASING
  - name: Back Yard Water Valve Battery
    id: sensor.back_yard_water_valve_battery
    area: garden
    device: back_yard_water_valve
    state: '75'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
# name: test_build[tests/homes/washer-example.yaml]
  '''
  ---
  areas:
  - name: Kitchen
    id: kitchen
  devices:
  - name: Lg Washer
    id: lg_washer
    area: kitchen
  entities:
  - name: Lg Washer Status
    id: sensor.lg_washer_status
    area: kitchen
    device: lg_washer
    state: idle
  - name: Lg Washer Remaining_time
    id: sensor.lg_washer_remaining_time
    area: kitchen
    device: lg_washer
    state: '0'
    attributes:
      device_class: sensor.SensorDeviceClass.DURATION
  - name: Lg Washer
    id: event.lg_washer
    area: kitchen
    device: lg_washer
    attributes:
      event_types:
      - started
      - finished
      - error
  
  
  '''
# ---
# name: test_build[tests/homes/weather-example.yaml]
  '''
  ---
  areas: []
  devices:
  - name: Home
    id: home
    info:
      manufacturer: AccuWeather
  entities:
  - name: Home
    id: weather.home
    device: home
    attributes:
      native_temperature_unit: "\xB0F"
      native_wind_speed_unit: mph
      condition: sunny
      native_temperature: 72
      humidity: 60
      native_wind_speed: 5
      daily_forecast:
      - cloudy
      - rainy
      hourly_forecast:
      - fair
      - fair
      - cloudy
      - cloudy
      twice_daily_forecast:
      - fair
      - cloudy
      - rainy
      - cloudy
  
  
  '''
# ---
# name: test_build[tests/homes/window-sensor-example.yaml]
  '''
  ---
  areas:
  - name: Family Room
    id: family_room
  devices:
  - name: Left Window Sensor
    id: left_window_sensor
    area: family_room
    info:
      manufacturer: Aqara
  entities:
  - name: Left Window Sensor
    id: binary_sensor.left_window_sensor
    area: family_room
    device: left_window_sensor
    state: false
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.WINDOW
  - name: Left Window Sensor Battery
    id: binary_sensor.left_window_sensor_battery
    area: family_room
    device: left_window_sensor
    attributes:
      device_class: binary_sensor.BinarySensorDeviceClass.BATTERY
  - name: Left Window Sensor Battery
    id: sensor.left_window_sensor_battery
    area: family_room
    device: left_window_sensor
    state: '90'
    attributes:
      device_class: sensor.SensorDeviceClass.BATTERY
      state_class: sensor.SensorStateClass.MEASUREMENT
      native_unit_of_measurement: '%'
  
  
  '''
# ---
